src/Tribe/Admin/Pages.php

```diff
@@ -0,0 +1,274 @@
+<?php
+namespace Tribe\Admin;
+
+/**
+ * Class Pages.
+ *
+ * @since TBD
+ */
+class Pages {
+	/**
+	 * Current page ID (or false if not registered with this controller).
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private $current_page = null;
+
+	/**
+	 * Registered pages
+	 * Contains information (breadcrumbs, menu info) about TEC admin pages.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private $pages = [];
+
+	/**
+	 * Get registered pages.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_pages() {
+		/**
+		 * Filters the list of registered TEC admin pages.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $pages The pages.
+		 */
+		$pages = apply_filters( 'tec_admin_pages', $this->pages );
+
+		return $pages;
+	}
+
+	/**
+	 * Adds a page to `tec-admin`.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $options {
+	 *   Array describing the page.
+	 *
+	 *   @type string      id           Id to reference the page.
+	 *   @type string      title        Page title. Used in menus and breadcrumbs.
+	 *   @type string|null parent       Parent ID. Null for new top level page.
+	 *   @type string      path         Path for this page, full path in app context; ex /analytics/report
+	 *   @type string      capability   Capability needed to access the page.
+	 *   @type string      icon         Icon. Dashicons helper class, base64-encoded SVG, or 'none'.
+	 *   @type int         position     Menu item position.
+	 *   @type int         order        Navigation item order.
+	 * }
+	 *
+	 * @return string $page The resulting page's hook_suffix.
+	 *
+	 */
+	public function register_page( $options ) {
```

Comment:
Either type-hint `$options` as `array` or add a check to handle the case where it's not an array.